### PR TITLE
Feature/use tmpdir gatk

### PR DESCRIPTION
**Pull request by ellendejong**
- Use --tmp-dir GATK4 tasks
- Use --TMP_DIR GATK4 - Picard tasks.
- Use java tmpdir (picard + gatk)

### CODE REVIEW

File: GATK/3.8-1-0-gf15c1c3ef/CombineVariants.nf [hunk 1]
```diff
@@ -14,7 +14,7 @@ process CombineVariants {
     script:
         def input_files = vcf_files.collect{"$it"}.join(" -V ")
         """
-        java -Xmx${task.memory.toGiga()-4}G -jar ${params.gatk_path} -T CombineVariants --reference_sequence ${params.genome} -V ${input_files} --out ${analysis_id}.vcf ${params.optional}
+        java -Xmx${task.memory.toGiga()-4}G -Djava.io.tmpdir=\$TMPDIR -jar ${params.gatk_path} -T CombineVariants --reference_sequence ${params.genome} -V ${input_files} --out ${analysis_id}.vcf ${params.optional}
```
Comment (rernst) [inline GATK/3.8-1-0-gf15c1c3ef/CombineVariants.nf:17]: Also split on multiple lines?

File: GATK/4.1.3.0/MergeVCFs.nf [hunk 2]
```diff
@@ -20,7 +20,8 @@ process MergeVCFs {
         gatk --java-options "-Xmx${task.memory.toGiga()-4}g -Djava.io.tmpdir=\$PWD" \
         SortVcf \
         --INPUT $vcfs \
-        --OUTPUT ${id}${ext}
+        --OUTPUT ${id}${ext} \
+        --TMP_DIR \$TMPDIR
```
Comment (rernst) [inline GATK/4.1.3.0/MergeVCFs.nf:24]: Gaat dit goed in combinatie met met: `-Djava.io.tmpdir=\$PWD"`

File: GATK/4.1.3.0/SplitNCigarReads.nf [hunk 3]
```diff
@@ -13,7 +13,7 @@ process SplitNCigarReads {
 
     script:
         """
-        gatk --java-options "-Xmx${task.memory.toGiga()-4}g" \
+        gatk --java-options "-Xmx${task.memory.toGiga()-4}g -Djava.io.tmpdir=\$TMPDIR" \
```
Comment (rernst) [inline GATK/4.1.3.0/SplitNCigarReads.nf:16]: `--tmp-dir` vergeten?

**Review by rernst — CHANGES_REQUESTED** [review]
Some small inconsistencies. Did you check Picard behaviour? Since it is also developed by broad.

File: GATK/4.1.3.0/SplitNCigarReads.nf (hunk 3 above)
Comment (ellendejong) [reply GATK/4.1.3.0/SplitNCigarReads.nf:16]: configured as `$PWD`, will change.

**Comment by ellendejong** [comment]
> Some small inconsistencies. Did you check Picard behaviour? Since it is also developed by broad.

Added java tmpdir to PICARD.

**Review by rernst — APPROVED** [review]

**Pull request merged**